Add Puppet enablement procedure (#890)
theforeman/foreman-documentation: guides/common/modules/proc_enabling-puppet.adoc

> :_module-type: PROCEDURE
> 

Reviewer: ```suggestion
```

I would delete the first two lines. If, at some point, we/you need metadata in each procedure, we should figure out a way to do this in one go.

[id="proc_enabling-puppet_{context}"]
= Enabling Puppet Integration with {Project}

A fresh installation of {Project} does not have any Puppet integration configured by default.
You need to enable the integration as is appropriate for your situation.

> {Project} can be configured to manage and deploy Puppet Server on the {ProjectServer} or on a {SmartProxy}.

Reviewer 2: ```suggestion
{Project} can be configured to manage and deploy Puppet Server on {ProjectServer} or on {SmartProxy}.
```
I know it seems unnatural but there is a convention for Satellite in particular not to use the article before Satellite and Capsule, so we don't use it.

:_module-type: PROCEDURE

> [id="proc_enabling-puppet_{context}"]

Reviewer: ```suggestion
[id="Enabling_Puppet_Integration_{context}"]
```

We usually set the anchor to the title + `_{context}`. Unfortunately, we cannot use two attributes in anchors.

= Enabling Puppet Integration with {Project}

> [role="_abstract"]

Reviewer: What does this mean/do? I have only seen this in `guides/common/modules/proc_synchronizing-ansible-collections.adoc` beforce.  I don't think we need this/rely on this. Do you need this for downstream?

Same goes for line one.

If you don't know for sure why, then please remove both instances.

> A fresh installation of {Project} does not have any Puppet integration configured by default.

Reviewer: ```suggestion
By default, {Project} does not ...
```

Please reword.

You need to enable the integration as is appropriate for your situation.
{Project} can be configured to manage and deploy Puppet Server on {ProjectServer} or on {SmartProxy}.

> Additionally, Puppet Server can be deployed externally to {Project} and integrated for reporting, facts and external node classification (ENC).

Reviewer: Please reword `can be deployed` > `you can deploy...`

.Prerequisites

> 

Reviewer: ```suggestion
```

* You installed and configured {Project}.

.Procedure

> 

Reviewer: ```suggestion
```

> . Enable Puppet on {ProjectServer} before you enable it on {SmartProxy}s:

Reviewer 2: ```suggestion
. Enable Puppet on {ProjectServer} before you enable it on {SmartProxies}:
```

+
[options="nowrap" subs="+quotes,attributes"]
----
# {foreman-installer} --enable-foreman-plugin-puppet \
--enable-foreman-cli-puppet \
--foreman-proxy-puppet true \
--foreman-proxy-puppetca true \

> --foreman-proxy-content-puppet true \

Reviewer 3: This should be hidden behind ifdef and only apply to Katello and its derivatives.

Reviewer 3: The background is that content is a Katello feature. That means any installer option that starts with `--foreman-proxy-content-` is just for Katello. So in this case:
```suggestion
 ifdef::katello,satellite,orcharhino[]
--foreman-proxy-content-puppet true \
 endif::[]
```

--enable-puppet \
--puppet-server true \
--puppet-server-foreman-ssl-ca /etc/pki/katello/puppet/puppet_client_ca.crt \
--puppet-server-foreman-ssl-cert /etc/pki/katello/puppet/puppet_client.crt \

> --puppet-server-foreman-ssl-key /etc/pki/katello/puppet/puppet_client.key

Reviewer 3: This should be hidden behind `ifdef` and only apply to Katello and its derivatives.

Reviewer 3: Here the background is that Katello has a different SSL certificate setup than regular Foreman. Regular Foreman uses Puppet's certificates so it isn't needed to override this:
```suggestion
ifdef::katello,satellite,orcharhino[]
--puppet-server-foreman-ssl-ca /etc/pki/katello/puppet/puppet_client_ca.crt \
--puppet-server-foreman-ssl-cert /etc/pki/katello/puppet/puppet_client.crt \
--puppet-server-foreman-ssl-key /etc/pki/katello/puppet/puppet_client.key
endif::[]
```

----

> 

Reviewer: ```suggestion
```

> . Enable Puppet on {SmartProxy}s:

Reviewer: ```suggestion
. Enable Puppet on {SmartProxies}:
```

We have a special attribute for that: https://github.com/theforeman/foreman-documentation/blob/master/guides/common/attributes-base.adoc#L95

+
[options="nowrap" subs="+quotes,attributes"]
----
# {foreman-installer} --foreman-proxy-puppet true \
--foreman-proxy-puppetca true \

> --foreman-proxy-content-puppet true \

Reviewer 3: This should be hidden behind ifdef and only apply to Katello and its derivatives.

--enable-puppet \
--puppet-server true \
--puppet-server-foreman-ssl-ca /etc/pki/katello/puppet/puppet_client_ca.crt \
--puppet-server-foreman-ssl-cert /etc/pki/katello/puppet/puppet_client.crt \

> --puppet-server-foreman-ssl-key /etc/pki/katello/puppet/puppet_client.key

Reviewer 3: This should be hidden behind ifdef and only apply to Katello and its derivatives.

----

> 
> .Verification
> 
> . Start each step with an active verb.
> 
> . Include one command or action per step.
> 
> . Use an unnumbered bullet (*) if the procedure includes only one step.

Reviewer: ```suggestion
```

I have no idea what this means; looks like a template.
